src/connection.js

```diff
@@ -663,6 +717,12 @@ class Connection extends EventEmitter {
     if (this.state === this.STATE.REROUTING) {
       this.debug.log('Rerouting to ' + this.routingData.server + ':' + this.routingData.port);
       return this.dispatchEvent('reconnect');
+    } else if (this.state === this.STATE.TRANSIENT_FAILURE_RETRY) {
+      const server = this.routingData ? this.routingData.server : this.server;
+      const port = this.routingData ? this.routingData.port : this.config.options.port;
+      this.debug.log('Retry after transient failure connecting to ' + server + ':' + port);
+
+      return this.dispatchEvent('retry');
```

Comment:
Why not just call `createRetryTimer()` here? `dispatchEvent()` is executed synchronously.